addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessResource.java

```diff
@@ -74,30 +74,65 @@ public NPMContentAccessResource( final ContentAccessHandler handler )
-     *        /{packageName}/package.json, /{packageName}/{version}/package.json, /{packageName}/{version}/package.tgz.
-     * @param uriInfo
-     * @param request
-     * @return
-     */
     @Override
-    @ApiOperation( "Store NPM artifact content under the given artifact store (type/name) and path." )
+    @ApiOperation( "Store NPM artifact content under the given artifact store (type/name) and packageName." )
     @ApiResponses( { @ApiResponse( code = 201, message = "Content was stored successfully" ), @ApiResponse( code = 400, message = "No appropriate storage location was found in the specified store (this store, or a member if a group is specified)." ) } )
     @PUT
-    @Path( "/{path: (.+)?}" )
+    @Path( "/{packageName}" )
     public Response doCreate( final @ApiParam( allowableValues = "hosted,group,remote", required = true ) @PathParam( "type" ) String type,
-                              final @ApiParam( required = true ) @PathParam( "name" ) String name, final @PathParam( "path" ) String path, final @Context UriInfo uriInfo,
+                              final @ApiParam( required = true ) @PathParam( "name" ) String name,
+                              final @PathParam( "packageName" ) String packageName, final @Context UriInfo uriInfo,
                               final @Context HttpServletRequest request )
     {
+        EventMetadata eventMetadata = new EventMetadata();
+        eventMetadata.set( STORAGE_PATH, Paths.get( packageName, PACKAGE_JSON ).toString() );
+
+        return handler.doCreate( NPM_PKG_KEY, type, name, packageName, request, eventMetadata,
+                                 () -> uriInfo.getBaseUriBuilder()
+                                              .path( getClass() )
+                                              .path( packageName )
+                                              .build( NPM_PKG_KEY, type, name ) );
+    }
+
+    @ApiOperation( "Store NPM artifact content under the given artifact store (type/name), packageName and version." )
+    @ApiResponses( { @ApiResponse( code = 201, message = "Content was stored successfully" ), @ApiResponse( code = 400,
+                                                                                                            message = "No appropriate storage location was found in the specified store (this store, or a member if a group is specified)." ) } )
+    @PUT
+    @Path( "/{packageName}/{version}" )
+    public Response doCreate(
+            final @ApiParam( allowableValues = "hosted,group,remote", required = true ) @PathParam( "type" )
+                    String type, final @ApiParam( required = true ) @PathParam( "name" ) String name,
+            final @PathParam( "packageName" ) String packageName, final @PathParam( "version" ) String version,
+            final @Context UriInfo uriInfo, final @Context HttpServletRequest request )
+    {
+        final String path = Paths.get( packageName, PATH_SPLITER, version ).toString();
```

Comment:
Will this result in creation of a path: `jquery//1.3.1`? That seems strange...

Is this for writing a package.json file or tgz content?